lib/SILOptimizer/Mandatory/TFCanonicalizeCFG.cpp

```diff
@@ -400,8 +416,30 @@ void SingleExitLoopTransformer::initialize() {
     }
   }
 
-  // Exit Blocks
-  loop->getExitBlocks(exitBlocks);
+  // Compute common exit block if needed.
+  if (TFNoUndefsInSESE) {
+    ensureSingleExitBlock();
+  }
+
+  // Initialize Exit Blocks
+  {
+    SmallVector<SILBasicBlock*, 8> exitBlockList;
+    // This API returns duplicates if there are multiple exiting edges to the
+    // same exit block. Dedup them here.
+    loop->getExitBlocks(exitBlockList);
+    exitBlocks.insert(exitBlockList.begin(), exitBlockList.end());
+    assert(!exitBlocks.empty() && "A loop has no exit blocks.");
+    if (TFNoUndefsInSESE) {
+      SmallPtrSet<SILValue, 8> exitArgs;
+      for (const SILBasicBlock *bb : exitBlocks) {
```

Comment:
in this case, should we assert `exitBlocks.size() == 1`?

Reply:
We will need to implement cloning for that.